tree_performance/benchmarks/numba_iterative.py

```diff
@@ -0,0 +1,74 @@
+import tskit
+import util
+import functools
+import numba
+import numpy as np
+import msprime
+
+VECTORISED = False
+
+@numba.njit()
+def _iterative_hartigan_parsimony(genotypes, samples, left_child, parent, postorder, preorder, root):
+    num_nodes = len(postorder)
+    num_alleles = np.max(genotypes) + 1
+    optimal_set = np.zeros((num_nodes, num_alleles), dtype=np.int8)
+    # Simple version assuming non missing data and one root
+    for allele, u in zip(genotypes, samples):
+        optimal_set[u, allele] = 1
+
+    allele_count = np.zeros((num_nodes, num_alleles), dtype=np.int32)
+    for node_j in postorder:
+        if left_child[node_j] != tskit.NULL:
```

Comment:
Probably better to hard-code -1 here. I have a feeling accessing a module variable like this could hurt perf? Although it's compiling down to nopython, so probably not.

Easily tested I guess.